core/src/Platforms/Android/SystemWebview/AuthenticationActivity.cs

```diff
@@ -135,7 +135,15 @@ protected override void OnResume()
                 browserIntent.SetData(Uri.Parse(_requestUrl));
                 browserIntent.SetPackage(chromePackage);
                 browserIntent.AddCategory(Intent.CategoryBrowsable);
-                StartActivity(browserIntent);
+                try
+                {
+                    StartActivity(browserIntent);
+                }
+                catch(ActivityNotFoundException ex)
+                {
+                    throw new MsalClientException(MsalClientException.ChromeDisabledError,
+                        "Chrome is disabled on the device, cannot proceed with authentication " + ex.Message);
```

Comment:
>"Chrome is disabled on the device, cannot proceed with authentication " [](start = 24, length = 71)

const?t like in ohter places and similar to the ID of the exception. Also consider adding the ex as the inner exception. ... this can be a factory method on the exception class.

Reply:
not sure I see this as done... not what I expected at least.

---
In reply to: [193146713](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1072#discussion_r193146713) [](ancestors = 193146713,192980110)

Reply:
I'll make the other changes as well

---
In reply to: [193172792](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1072#discussion_r193172792) [](ancestors = 193172792,193146713,192980110)